### PR TITLE
[DEMO] Add text recognition to text_detection_demo

### PR DESCRIPTION
**Pull request by LeonidBeynenson**

### CODE REVIEW

**Comment by LeonidBeynenson** [comment]
@snosov1:     
Sergei, this is the PR with the latest text detection/recognition changes.

**Comment by LeonidBeynenson** [comment]
2 @vmatveic:   FYI

**Comment by LeonidBeynenson** [comment]
Jenkins please retry a build

**Comment by LeonidBeynenson** [comment]
Jenkins please retry a build

**Comment by LeonidBeynenson** [comment]
Jenkins please retry a build

**Comment by LeonidBeynenson** [comment]
I have squashed the commits.

@grib0ed0v, @evgeny-izutov -- could you review

2 @snosov1 , @vmatveic:  Could you review and merge?

File: demos/text_detection_demo/main.cpp [hunk 1]
```diff
@@ -60,138 +71,211 @@ int main(int argc, char *argv[]) {
...
+                    std::chrono::steady_clock::time_point begin = std::chrono::steady_clock::now();
+                    points = floatPointsFromRotatedRect(rect);
+                    topLeftPoint(points, &top_left_point_idx);
+                    cropped_text = cropImage(image, points, text_recognition.input_size(), top_left_point_idx);
+                    std::chrono::steady_clock::time_point end = std::chrono::steady_clock::now();
+                    text_crop_time += std::chrono::duration_cast<std::chrono::microseconds>(end - begin).count();
+                } else {
+                    if (FLAGS_cc) {
+                        int w = image.cols * 0.05;
+                        int h = w * 0.5;
```
Comment (evgeny-izutov) [inline demos/text_detection_demo/main.cpp]: Possible warning without `static_cast`?

File: demos/text_detection_demo/main.cpp [hunk 2]
```diff
@@ -60,138 +71,211 @@ int main(int argc, char *argv[]) {
...
-        image.convertTo(image, CV_32F);
+                    std::chrono::steady_clock::time_point begin = std::chrono::steady_clock::now();
+                    res = CTCGreedyDecoder(output_data, kAlphabet, kPadSymbol, &conf);
+                    std::chrono::steady_clock::time_point end = std::chrono::steady_clock::now();
+                    text_recognition_postproc_time += std::chrono::duration_cast<std::chrono::microseconds>(end - begin).count();
 
-        cv::resize(image, image, cv::Size(input_info->getTensorDesc().getDims()[3],
-                   input_info->getTensorDesc().getDims()[2]));
+                    res = conf >= min_text_recognition_confidence ? res : "";
+                    num_found += !res.empty() ? 1 : 0;
```
Comment (evgeny-izutov) [inline demos/text_detection_demo/main.cpp]: Is it better to replace on cast operation?

File: demos/text_detection_demo/main.cpp (hunk 1 above)
Comment (LeonidBeynenson) [reply demos/text_detection_demo/main.cpp]: Ok, agree

File: demos/text_detection_demo/main.cpp (hunk 2 above)
Comment (LeonidBeynenson) [reply demos/text_detection_demo/main.cpp]: I think this is not so important...

File: demos/text_detection_demo/main.cpp (hunk 1 above)
Comment (LeonidBeynenson) [reply demos/text_detection_demo/main.cpp]: Done

**Comment by LeonidBeynenson** [comment]
Jenkins please retry a build

File: demos/text_detection_demo/main.cpp (hunk 2 above)
Comment (evgeny-izutov) [reply demos/text_detection_demo/main.cpp]: Ok

**Comment by LeonidBeynenson** [comment]
Jenkins please retry a build

**Pull request merged**